src/tasks/AppleAppBuilder/Templates/runtime.m

```diff
@@ -261,8 +253,34 @@ static int32_t load_icu_data ()
     const char* bundle = get_bundle_path ();
     chdir (bundle);
 
+    char icu_dat_path[1024];
+    int res;
+
+    res = snprintf (icu_dat_path, sizeof (icu_dat_path) - 1, "%s/%s", bundle, "icudt.dat");
+    assert (res > 0);
+
     // TODO: set TRUSTED_PLATFORM_ASSEMBLIES, APP_PATHS and NATIVE_DLL_SEARCH_DIRECTORIES
-    monovm_initialize(0, NULL, NULL);
+#if !INVARIANT_GLOBALIZATION
+    const char* appctx_keys[3];
+    appctx_keys[0] = "RUNTIME_IDENTIFIER";
+    appctx_keys[1] = "APP_CONTEXT_BASE_DIRECTORY";
+    appctx_keys[2] = "ICU_DAT_FILE_PATH";
+
+    const char* appctx_values[3];
+    appctx_values[0] = APPLE_RUNTIME_IDENTIFIER;
+    appctx_values[1] = bundle;
+    appctx_values[2] = icu_dat_path;
+    monovm_initialize(3, appctx_keys, appctx_values);
```

Comment:
You can do this part a little bit cleaner for example see 

https://github.com/lambdageek/monovm-embed-sample/blob/41b06385f80e54a3b24af9c4ad4c79cc24dff3e9/src/native/main.c#L59-L62

Use `sizeof(appctx_values)/sizeof(appctx_values[0])` for the number of properties - and then move the call to `monovm_initialize` below the `#ifdef`.

Potentially you could do something like this, too:
```
const char *appctx_keys[] = {
  "key1", 
#ifndef INVARIANT_GLOBALIZATION
  "key2",
#endif
  "key3",
};
const char *appctx_values [] = {
  "value1",
#ifndef INVARIANT_GLOBALIZATION
  "value2",
#endif
  "value3",
};
```

although i'm not sure if duplicating the ifdefs is a great idea.  might get key/value mismatches.